core/src/main/scala/org/http4s/headers/X-B3-Flags.scala

```diff
@@ -0,0 +1,59 @@
+package org.http4s
+package headers
+
+import org.http4s.parser.HttpHeaderParser
+import org.http4s.util.Writer
+import org.http4s.{ Header, HeaderKey, ParseResult }
+
+import scalaz.ISet
+
+object `X-B3-Flags` extends HeaderKey.Internal[`X-B3-Flags`] with HeaderKey.Singleton {
+  override def parse(s: String): ParseResult[`X-B3-Flags`] =
+    HttpHeaderParser.X_B3_FLAGS(s)
+
+  sealed trait Flag extends Product with Serializable {
+    def longValue: Long
+  }
+
+  object Flag {
+    case object Debug extends Flag {
+      override def longValue: Long = 1 << 0
+    }
+    case object SamplingSet extends Flag {
+      override def longValue: Long = 1 << 1
+    }
+    case object Sampled extends Flag {
+      override def longValue: Long = 1 << 2
+    }
+  }
+
+  def fromLong(flagBits: Long): `X-B3-Flags` = {
```

Comment:
I don't see anything wrong with this as far as logic, but it feels a bit more convoluted than it needs to be. I suspect there would be a lot less code and it would be easier to read using a local `var` and three `if` statements. I'm not sure on this final point, but I think this will allocate quite a few lambda functions.